code/datums/mind.dm

```diff
@@ -243,15 +237,12 @@
 
 	for(var/datum/antagonist/A as anything in antag_datums)
 		all_objectives += A.objective_holder.get_objectives() // Add all antag datum objectives.
-		if(include_team)
-			var/datum/team/team = A.get_team()
-			if(team) // have to make asure a team exists here, team?. does not work below because it will add the null to the list
-				all_objectives += team.objective_holder.get_objectives() // Get all of their teams' objectives
-
-	// For custom non-antag role teams
-	if(include_team && LAZYLEN(teams))
-		for(var/datum/team/team as anything in teams)
-			all_objectives += team.objective_holder.get_objectives()
+		if(!include_team)
+			continue
+
+		var/datum/team/team = A.get_team()
+		if(team) // have to make asure a team exists here, team?. does not work below because it will add the null to the list
+			all_objectives += team.objective_holder.get_objectives() // Get all of their teams' objectives
```

Comment:
imo this refactor makes it more unreadable, and it wont grant any performance increases